src/unity/python/turicreate/visualization/plot.py

```diff
@@ -139,16 +140,19 @@ def save(self, filepath, include_data=True):
             # save as png/svg, but json first
             spec = self._get_vega(include_data = True)
             extension = filepath[-3:]
-            with open("temp_file.vg.json", 'w') as fp:
+            temp_file_tuple = _mkstemp()
+            temp_file_path = temp_file_tuple[1]
+            with open(temp_file_path, 'w') as fp:
                 _json.dump(spec, fp)
-            dirname = os.path.dirname(__file__)
-            relative_path_to_vg2png_vg2svg = "../../../../../../node_modules/vega/bin/vg2" + extension
-            absolute_path_to_vg2png_vg2svg = os.path.join(dirname, relative_path_to_vg2png_vg2svg)
-            os.system("node " + absolute_path_to_vg2png_vg2svg + " temp_file.vg.json " + filepath)
+            dirname = _os.path.dirname(__file__)
+            relative_path_to_vg2png_vg2svg = "../node_modules/vega/bin/vg2" + extension
+            absolute_path_to_vg2png_vg2svg = _os.path.join(
+                dirname, relative_path_to_vg2png_vg2svg)
+            _os.system("node " + absolute_path_to_vg2png_vg2svg + " " + temp_file_path + " " + filepath)
```

Comment:
Before we call into an external `node` binary, we should verify that one exists in the user's path. If this call is going to fail, we should instead present a nice error message to the user telling them to install node (and linking to the Node download page).

Reply:
That can come in a future PR.